server/utils.go

```diff
@@ -0,0 +1,30 @@
+package server
+
+import (
+	"regexp"
+	"strings"
+)
+
+// SplitExternalHosts splits a string containing external hostnames by comma and/or newline delimiters.
+// It trims whitespace around each hostname and filters out empty strings.
+// Examples:
+//   - "host1.com,host2.com" -> ["host1.com", "host2.com"]
+//   - "host1.com, host2.com" -> ["host1.com", "host2.com"]
+//   - "host1.com\nhost2.com" -> ["host1.com", "host2.com"]
+//   - "host1.com,\nhost2.com" -> ["host1.com", "host2.com"]
+func SplitExternalHosts(s string) []string {
+	// Use regexp to split on either comma or newline
+	re := regexp.MustCompile(",|\n")
```

Comment:
The MustCompile made me realize this would be best declared as a const. Then it can be slightly optimized.